src/util/http/HttpClient.h

```diff
@@ -58,7 +60,7 @@ class HttpClientImpl {
  private:
   // The connection stream and associated objects. See the implementation of
   // `openStream` for why we need all of them, and not just `stream_`.
-  boost::asio::io_context io_context_;
+  boost::asio::thread_pool executor_{1};
```

Comment:
Currently this means a new thread is spawned for every Service operation. Not ideal. Perhaps making this static would be better?